dashboard/src/components/Common/XpubField/XpubField.jsx

```diff
@@ -39,29 +60,46 @@ class XpubField extends React.Component {
             <label key={`key-${this.props.index}-option-${key}`}>
               <input type='radio'
                 name={`keys-${this.props.index}`}
+                onChange={typeOnChange}
+                checked={key == typeProps.value}
                 value={key}
-                checked={key == this.state.selectedType}
-                onChange={radioChanged}
               />
               {methodOptions[key]}
             </label>
           )}
         </div>
 
-        {this.state.selectedType == 'mockhsm' &&
-          <SelectField options={this.props.mockhsmKeys}
+        {typeProps.value == 'mockhsm' &&
+          <SelectField options={mockhsmKeys}
             valueKey='xpub'
             labelKey='label'
-            fieldProps={this.props.fieldProps} />
+            fieldProps={{...valueProps, onChange: valueOnChange}} />
         }
 
-        {this.state.selectedType == 'provide' &&
-          <TextField key={this.props.index} fieldProps={this.props.fieldProps} />}
+        {typeProps.value == 'provide' &&
+          <TextField
+            fieldProps={{...valueProps, onChange: valueOnChange}}
+            placeholder='Enter Xpub' />}
+
+        {typeProps.value == 'generate' &&
+          <TextField
+            fieldProps={{...valueProps, onChange: valueOnChange}}
```

Comment:
It doesn't propoagte because the `type` and `value` `onChange` functions work in tandem to sync with the component state for each option individually.